klog_test.go

```diff
@@ -1365,6 +1365,74 @@ func TestErrorSWithLogr(t *testing.T) {
 	}
 }
 
+func TestCallDepthLogr(t *testing.T) {
+	logger := &callDepthTestLogr{
+		callDepth: -1,
+	}
```

Comment:
```suggestion
	logger := &callDepthTestLogr{}
	logger.resetCallDepth()
```
Let's not set depth in two different places.